runtime/compiler/optimizer/J9TransformUtil.cpp

```diff
@@ -933,6 +947,20 @@ J9::TransformUtil::foldStaticFinalFieldAssumingProtection(TR::Compilation *comp,
    return false;
    }
 
+/** \brief
+ *     Compile time query that answers if a direct load for static can be folded by compiler.
+ *
+ *  \param comp
+ *     The compilation object.
+ *
+ *  \param node
+ *     The node which is a load of a static final field.
+ *
+ *  \return
+ *    True TR_yes if the field is whitelisted and can be folded.
+ *    True TR_maybe if the field is a generic static final and can be folded with protection.
+ *    True TR_no if the field is cannot be folded due to being unresolved, owning class uninitialized, etc.
```

Comment:
another important case for no is that the static final has already been written post-initialization...

Reply:
Added in comments.